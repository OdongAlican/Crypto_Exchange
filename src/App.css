.App {
  background-color: #F8F8F8;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.input-sections {
  background-color: #fff;
  padding: 10px;
  position: relative;
}

.input-sections-lower {
  display: flex;
}

i {
  cursor: pointer;
}

input,
select {
  height: 35px;
  width: 300px;
  margin-bottom: 20px;
  border-radius: 3px;
  outline: none;
  border: 1px solid #DDDDDD;
  padding: 0 10px;
}

.icon-section {
  background-color: #2F71D9;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 35px;
  padding: 0 10px;
  margin: 0 5px;
  color: #fff;
  border-radius: 3px;
}

.information-section {
  display: flex;
  justify-content: center;
}

.information-section div {
  margin-left: 20px;
}

.display-rates {
  position: absolute;
  top: -100px;
  right: 0;
  padding: 20px;
  background-color: #fff;
}