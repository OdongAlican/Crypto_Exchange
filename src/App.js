import { useEffect, useState } from 'react';
import axios from 'axios';
import './App.css';

const App = () => {
  const [coin, setCoin] = useState('BTC');
  const [currency, setCurrency] = useState('USD');
  const [value, setSetValue] = useState(1);
  const [currentCoin, setCurrentCoin] = useState('');
  const [currentCurrency, setCurrentCurrency] = useState('');
  const [amount, setAmount] = useState('');
  const API = '95CF4C23-5C1F-467A-910C-F6F9DBCEBC79';
  // const API2 = "9B4DA46A-CA8D-4D85-924D-7F2FF3F413C7";

  const APIREQUEST = axios.create({
    baseURL: 'https://rest.coinapi.io/v1/exchangerate',
    headers: { 'X-CoinAPI-Key': API,}
  });

  const setCoinName = coin => {
    if(coin === 'BTC') {setCurrentCoin('Bitcoin (BTC)')}
    if (coin === 'DGD') { setCurrentCoin('Dodge Coin (DGD)')} 
    if (coin === 'ETH') {setCurrentCoin('Etherium (ETH)')}
  };

  const setCurrencyName = currency => {
    if(currency === 'USD') {setCurrentCurrency('United Sates Dollars "$" (USD)')}
    if (currency === 'GBP') { setCurrentCurrency('Great British Pound "£" (GBP)')} 
    if (currency === 'EUR') {setCurrentCurrency('Euros "€" (EUR)')}
  };

  useEffect(()=> {
    if(coin && currency) {   
      APIREQUEST.get(`/${coin}/${currency}`).then(response => {
        setAmount((response?.data?.rate) * value);
        setCoinName(coin);
        setCurrencyName(currency);
      }
        ).catch(error => console.log(error));
    }
  }, [coin,currency, value]);
 
  return (
    <div data-testid="appCoin" className="App">
      <div className="input-sections">
        <div className="input-sections-top">
          <input aria-label="value-input" value={value} type="number" onChange={ e => {
            if(e.target.value < 0) 
            {
              alert('We cannot exchange for values less than 0!!')
              return setSetValue(1);              
            } 
              setSetValue(e.target.value);
            }
            }/>
        </div>
        <div className="input-sections-lower">
          <select name="coin" value={coin} onChange={e => setCoin(e.target.value)}>
            <option value="BTC">Bitcoin (BTC)</option>
            <option value="DGD">Dodgecoin (DGD)</option>
            <option value="ETH">Etherium (ETH)</option>
          </select>
          <div className="icon-section">
          <i className="fas fa-exchange-alt" />
          </div>
          <select name="currency" value={currency} onChange={e => setCurrency(e.target.value)}>
            <option value="USD">United States Dollar "$" (USD)</option>
            <option value="GBP">British Pounds "£" (GBP)</option>
            <option value="EUR">Euro "€" (EUR)</option>
          </select>
        </div>
            {
              !amount ? (
                <div className="information-section">Loading !!!</div>
              ) : (
                <div className="information-section">
                  <div>{value} {currentCoin}</div>
                  <div> = </div>
                  <div>{ amount && parseFloat(amount).toFixed(2)} {currentCurrency}</div>
                </div>
              )
            }
      </div>
    </div>
  );
}

export default App;
