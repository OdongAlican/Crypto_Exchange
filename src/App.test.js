import { render, fireEvent } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

const setup = () => {
  const utils = render(<App />)
  const input = utils.getByLabelText('value-input')
  return { input, ...utils }
}

it('It should not allow letters to be inputted', () => {
  const {input} = setup();
  expect(input.value).toBe('1');
  fireEvent.change(input, {target: {value: 'Random String'}})
  expect(input.value).toBe('');
  fireEvent.change(input, {target: {value: 12}})
  expect(input.value).toBe("12");
});

it('It renders App component correctly', () => {
  const { getAllByTestId } = render(<App />);
  const result = getAllByTestId('appCoin');
  expect(result[0]).toHaveClass('App');
  expect(result[0]).toHaveTextContent('Bitcoin');
  expect(result[0]).not.toHaveTextContent('Radom Crypto');
})

it('matches App snapshot', () => {
  const appointment = renderer.create(<App />);
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
