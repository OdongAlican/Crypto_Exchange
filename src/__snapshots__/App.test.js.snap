// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches App snapshot 1`] = `
<div
  className="App"
  data-testid="appCoin"
>
  <div
    className="input-sections"
  >
    <div
      className="input-sections-top"
    >
      <input
        aria-label="value-input"
        onChange={[Function]}
        type="number"
        value={1}
      />
    </div>
    <div
      className="input-sections-lower"
    >
      <select
        name="coin"
        onChange={[Function]}
        value="BTC"
      >
        <option
          value="BTC"
        >
          Bitcoin (BTC)
        </option>
        <option
          value="DGD"
        >
          Dodgecoin (DGD)
        </option>
        <option
          value="ETH"
        >
          Etherium (ETH)
        </option>
      </select>
      <div
        className="icon-section"
        onClick={[Function]}
      >
        <i
          className="fas fa-exchange-alt"
        />
      </div>
      <select
        name="currency"
        onChange={[Function]}
        value="USD"
      >
        <option
          value="USD"
        >
          United States Dollar "$" (USD)
        </option>
        <option
          value="GBP"
        >
          British Pounds "£" (GBP)
        </option>
        <option
          value="EUR"
        >
          Euro "€" (EUR)
        </option>
      </select>
    </div>
    <div
      className="information-section"
    >
      <div>
        1
         
        BTC
      </div>
      <div>
         = 
      </div>
      <div>
        0
         
        USD
      </div>
    </div>
  </div>
</div>
`;
